src/posix/platform/radio.cpp

```diff
@@ -146,61 +141,54 @@ void Radio::Init(void)
         SuccessOrDie(otPlatRadioSetRegion(gInstance, regionCode));
     }
 
-#if OPENTHREAD_POSIX_CONFIG_MAX_POWER_TABLE_ENABLE
-    maxPowerTable = mRadioUrl.GetValue("max-power-table");
-    if (maxPowerTable != nullptr)
+    ProcessMaxPowerTable(aRadioUrl);
+
+#if OPENTHREAD_CONFIG_PLATFORM_RADIO_COEX_ENABLE
     {
-        constexpr int8_t kPowerDefault = 30; // Default power 1 watt (30 dBm).
-        const char      *str           = nullptr;
-        uint8_t          channel       = ot::Radio::kChannelMin;
-        int8_t           power         = kPowerDefault;
-        otError          error;
-
-        for (str = strtok(const_cast<char *>(maxPowerTable), ","); str != nullptr && channel <= ot::Radio::kChannelMax;
```

Comment:
OpenThread is going to be used in multi-thread environment. we should use `strtok_r` instead.

Reply:
Updated